### PR TITLE
Pileupcaller individuals together, based on library construction. #484

### PR DESCRIPTION
**Pull request by jfy133**
# nf-core/eager pull request

This makes a few structual changes to the pileupcaller module originally from @sc13-bioinf by @TCLamnidis . This now collects all in BAMs into a process for double-stranded and single-stranded samples respectively, as this is how downstream tools require input. Note a later PR maybe will be made with a separate python tool to further merge these.

This also adds additional flags, and documentation clarifications.

## PR checklist

- [ ] This comment contains a description of changes (with reason)
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If necessary, also make a PR on the [nf-core/eager branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/eager)
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --paired_end`).
- [ ] Make sure your code lints ([`nf-core lint .`](https://nf-co.re/tools)).
- [ ] Documentation in `docs` is updated
- [ ] `CHANGELOG.md` is updated
- [ ] `README.md` is updated

**Learn more about contributing:** [CONTRIBUTING.md](https://github.com/nf-core/eager/tree/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: docs/output.md [hunk 1]
```diff
@@ -407,7 +421,7 @@ The MultiQC DamageProfiler module misincorporation plots shows the percent frequ
 When looking at the misincorporation plots, keep the following in mind:
 
 - As few-base single-stranded overhangs are more likely to occur than long overhangs, we expect to see a gradual decrease in the frequency of the modifications from position 1 to the inside of the reads.
-- If your library has been **partially-UDG treated**, only the first one or two bases will display the misincorporation frequency.
+- If your library has been **partially-UDG treated**, only the first one or two bases will display the the misincorporation frequency.
```
Comment (apeltzer) [inline docs/output.md:424]:
```suggestion
- If your library has been **partially-UDG treated**, only the first one or two bases will display the misincorporation frequency.
```

**Comment by jfy133** [comment]
@TCLamnidis please confirm this is working and merge

**Comment by TCLamnidis** [comment]
Starting a run now. will report back

**Comment by TCLamnidis** [comment]
I get the following error now
```
Error executing process > 'sex_deterrmine (1)'

Caused by:
  Process `sex_deterrmine` input file name collision -- There are multiple input files for each of the following file names: AO2001_libmerged_rg_rmdup.bam, ERI004_libmerged_rg_rmdup.bam


Tip: view the complete command output by changing to the process work dir and entering the command `cat .command.out`
```

**Comment by jfy133** [comment]
This occurs because we genotype, and consider biological sex to be best performed, at a `Sample_Id` level,

However, as one wants to genotype SS and DS libraries of a sample _separately_ with the schema above they should be then considered separate samples, as otherwise they would normally be merged. However after talking with @TCLamnidis the two samples were specified with the same name. 

There isn't an easy fix here yet I think. Given SS libraries are still not widely used (just Leipzig and Jena), I would prefer to not invest time now until we get more feedback. I will however add more docs to make it clear that genotyping should happen on distinct samples! 

@TCLamnidis If you let me know if your local tests pass after giving different sample names to the SS/DS libs, then we can merge.

**Comment by jfy133** [comment]
**Reminder to self** before merging update JSON schema with extra flag added by Thiseas!

**Comment by TCLamnidis** [comment]
There was a conflict between ssDNA libraries and the transitions mode (skip/keep/missing). in pileupcaller these options are mutually exclusive, so I added a check that ignores the transitions mode for the ssDNA genotyping and uses the selected one for dsDNA genotyping. 
The run now correctly produced 2 datasets, one for ssDNA and one for dsDNA, each containing all relevant individuals.

**Comment by TCLamnidis** [comment]
In the future, I would prefer to default to using separate ds/ss bams for genotyping and a merged one for sexDet, to avoid the sexdet error, although as discussed with @jfy133 that setting might be a bit Jena/Leipzig specific since ssDNA libraries are not too common atm. It still seems like the more straight forward solution to me personally, instead of asking users to "hack" their input TSVs and change what a sampleID means. 
Having to edit the sampleIDs would especially mess with possibilities to automate TSV generation from LIMS databases etc (like pandora), as the scripts would then need to be aware if the same sample with different library setups is present in the resulting TSV, and edit the field accordingly.

**Comment by jfy133** [comment]
:+1: @TCLamnidis . The issue is we can go into an infinite level of complexity regarding giving a user thousands of ways to merge/genotype. This would actually be suitable for DSL2 but that will be a long way off. I guess the question is how much 'experiemntation' do we want nf-core/eager to allow vs 'production'. Mixing SS/DS is quite experimental in a sense. But like Thiseas says, something to discuss in the future.

Will merge once tests pass!

**Pull request merged**